### ISSUE TITLE
document server-side dry run is GA

### ISSUE THREAD

**Issue by julianvmodesto** [kubernetes/kubernetes#86526]
**What would you like to be added**:

Server-side dry-run is being promoted from beta to GA in v1.18.

For the server-side dry-run integration in kubectl, we are making a few changes:

- we're extending the `--dry-run` flag to support the options `client`, `server` and `none`
- we're deprecating no value for the `--dry-run` flag, such that setting a value will be required in the future
- we're deprecating the boolean values for the `--dry-run` flag and will remove these values in the future
- we're deprecating the `kubectl apply` `--server-dry-run` flag

We need to document these changes accordingly on the website and in comments and tests.

Documentation includes:

- [x] document server-side dry-run feature state as stable for v1.18 (update Hugo shortcode) https://github.com/kubernetes/website/pull/19549
- [x] dry-run auth requirements @julianvmodesto https://github.com/kubernetes/website/pull/18235
- [ ] document all API requests, including server-side dry-run, are single-resource requests https://github.com/kubernetes/kubernetes/issues/83562 https://github.com/kubernetes/kubernetes/issues/86026 https://github.com/kubernetes/website/pull/19550
- [x] deprecation notices for `--dry-run`: covered by release notes
- [x] deprecation notice for `--server-dry-run`: covered by release notes
- [x] all usage and examples on the website https://github.com/kubernetes/website/pull/19549
- [x] all usage in code comments and tests

**Why is this needed**:

Server-side dry-run is graduating to GA, so we need to document usage and examples for users for how to use the feature.

Because we're extending an existing flag and deprecating aspects of this existing feature, we need to have consistent documentation for the flag that reflects the changes.

https://github.com/kubernetes/kubernetes/issues/85652
GA umbrella issue: https://github.com/kubernetes/kubernetes/issues/68514
KEP: https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/0015-dry-run.md

cc @apelisse 
/sig cli
/sig api-machinery
/priority important-soon
/help

**Comment by bharaththiruveedula** [comment]
@julianvmodesto can you please assign this to me?

**Comment by julianvmodesto** [comment]
Yes we could use some help here!

There are many places we'll need to identify and update in documentation, comments and tests for this upcoming change for this release cycle. 

/assign @bharaththiruveedula

**Comment by apelisse** [comment]
Yeah, we can use help since every occurence of `--dry-run` needs to be updated on the website and everywhere it's documented or referenced. Thanks for your help @bharaththiruveedula

**Comment by LucianaMarques** [comment]
Hey there! Has any PR been opened on this? Can I also help?

**Comment by apelisse** [comment]
Hi @LucianaMarques , I'm not aware of any activity going on right now on this task, we'd really appreciate your help! feel free to ask questions too if anything needs to be clarified, thank you!

**Comment by bharaththiruveedula** [comment]
@LucianaMarques , sorry, I am working on it

**Comment by apelisse** [comment]
@bharaththiruveedula any update on the progress? We have 0 visibility here! :-)

**Comment by apelisse** [comment]
Also, feel free to join and talk on #wg-apply, I suspect there is enough work for the two of you, but we need to be able to know who's doing what.

**Comment by bharaththiruveedula** [comment]
@apelisse sorry for the late... will share the details by tomorrow in the slack

**Comment by apelisse** [comment]
Thanks, no problem!

**Comment by LucianaMarques** [comment]
@apelisse thank you so much for the heads up, I will join #wg-apply!

**Comment by julianvmodesto** [comment]
This issue is unblocked and could use some help! There are many places in kubernetes.io that we need to identify then update and we need to add deprecation warnings for the old values.

**Comment by julianvmodesto** [comment]
From the release team:

> * Docs PRs must be ready for review on Mon March 9
> * Docs must be complete, all PRs reviewed and ready to merge by Mon March 16

Help is appreciated! I'll have a PR for Monday if there's no one else working on this

**Comment by julianvmodesto** [comment]
I think we're good here. I'm tracking this PR as a follow-up that will be merged into the master branch for docs since it isn't v1.18-specific: https://github.com/kubernetes/website/pull/19550

/close
